src/connector/src/parser/mod.rs

```diff
@@ -169,7 +191,10 @@ impl OpAction for OpActionInsert {
 
     #[inline(always)]
     fn finish(writer: &mut SourceStreamChunkRowWriter<'_>) {
-        writer.op_builder.push(Op::Insert)
+        writer.op_builder.push(Op::Insert);
+        if let Some(offset) = writer.row_offset.clone() {
```

Comment:
By making `offset` a normal column, I guess we don't need hacks for this anymore as well.